### PR TITLE
Spinny hammer :)

### PR DESCRIPTION
**Pull request by SollyBunny**
Looks cool

I called the variable cl_spinny_hammer and can be toggled in settings

image

## Checklist

- [*] Tested the change ingame
- [*] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [*] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

https://github.com/user-attachments/assets/9a973471-bc77-4141-934c-93c96ef3fca2

### CODE REVIEW

**Comment by SollyBunny** [comment]
#9080

**Comment by BlaiZephyr** [comment]
Thank you for your first PR! :)

i have to say.. i am not totally against this :P it's a cosmetic thing and off by default.

the button seems out of place in the DDNet Tab and may be unneccessary in general - we have a few settings that dont have a toggleable button like `cl_old_gun_position` for example.

also the naming is not really the best, you'd want the naming to be clear and precise, so `SpinnyHammer` is a little off. ^^

since this is a cosmetic change it might not make it in here, but tater already offered to accept it into his client which many people use. in #9070 

best of luck!

**Comment by SollyBunny** [comment]
Well it makes the hammer spinny, HammerSpinsLikeOtherWeapons is a bit wordy. There is also a description.

What would you name it? @BlaiZephyr

**Comment by SollyBunny** [comment]
@def- doesnt have the same ring to it as spinny hammer but okay

**Comment by sjrc6** [comment]
It looks much more natural than I expected. definitely a nice feature.

**Comment by heinrich5991** [comment]
I'd prefer to keep optional visual gimmicks to third-party clients. Other maintainers seem to disagree, so don't weigh my opinion too much.

**Comment by heinrich5991** [comment]
Sorry, I didn't see that this is your first PR. Thanks for your first PR! :)

**Comment by l-ouis** [comment]
I also agree with heinrich on the visual gimmicks, but I think this could be made pretty enough if you clamp the rotation (so it still rotates, just a little less so it doesn't ever become horizontal [that makes it look odd])
